src/ng-select/ng-dropdown-panel.component.ts

```diff
@@ -135,13 +135,15 @@ export class NgDropdownPanelComponent implements OnInit, OnChanges, OnDestroy, A
     }
 
     ngAfterContentInit() {
-        this._whenContentReady().then(() => {
-            if (this.appendTo) {
-                this._appendDropdown();
-                this._handleDocumentResize();
-            }
-            this.updateDropdownPosition();
-        });
+        from(this._whenContentReady())
```

Comment:
It looks a bit confusing to wrap this promise into observable as there is only one event. You could solve this by issue by checking if destroy subject was closed.
```ts
this._whenContentReady().then(() => {
    if (this._destroy$.closed) {
        return;
    }
    if (this.appendTo) {
        this._appendDropdown();
        this._handleDocumentResize();
    }
    this.updateDropdownPosition();
});
```

Reply:
`.closed` indicates whether or not the subscription has been unsubscribed which in current implementation was not.
So I've added `.unsubscribe()` to `ngOnDestroy` otherwise your suggestion won't work.

Reply:
Good catch, thanks 👍